src/app/location-map/map-modal/map-modal.component.ts

```diff
@@ -1,26 +1,26 @@
-import { Component, OnInit, AfterViewInit, Input } from '@angular/core';
+import { AfterViewInit, Component } from '@angular/core';
 import { BsModalRef } from 'ngx-bootstrap/modal';
 
 @Component({
     selector: 'app-map-modal',
     templateUrl: './map-modal.component.html',
     styleUrls: ['./map-modal.component.css'],
 })
-export class MapModalComponent implements OnInit, AfterViewInit {
-    @Input() projectCoords = [];
-    @Input() rerenderMap;
+export class MapModalComponent implements AfterViewInit {
+    modalShown = false;
+    lat: number;
+    lng: number;
 
-    constructor(private bsModalRef: BsModalRef) { }
-
-    ngOnInit() {}
+    constructor(private bsModalRef: BsModalRef) {
+    }
 
-    ngAfterViewInit() {}
+    ngAfterViewInit() {
+        // Required is an issue with showing leaflet map before
+        // modal is completely rendered. https://stackoverflow.com/q/20400713/4636976
+        this.modalShown = true;
+    }
```

Comment:
This seems to be a better approach to this issue. 
You made it from outside with `this.bsModalRef.content.rerenderMap.invalidateSize();`, but it's complicated to follow and why you do this to the next programmer that comes after you. If you do something for some edge case (e.g. 3rd-party bug), please make a comment on why you do it.